myriad-scheduler/src/main/java/com/ebay/myriad/state/utils/ByteBufferSupport.java

```diff
@@ -0,0 +1,328 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ebay.myriad.state.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.mesos.Protos;
+import com.ebay.myriad.scheduler.NMProfile;
+import com.ebay.myriad.state.NodeTask;
+import com.google.gson.Gson;
+import com.google.protobuf.GeneratedMessage;
+
+/**
+* ByteBuffer support for the Serialization of the StoreContext
+*/
+public class ByteBufferSupport {
+
+  public static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
+  public static final String UTF8 = "UTF-8";
+  public static final byte[] ZERO_BYTES = new byte[0];
+  private static Gson gson = new Gson();
+
+  public static void addByteBuffers(List<ByteBuffer> list,
+    ByteArrayOutputStream bytes) throws IOException {
+    // If list, add the list size, then the size of each buffer followed by the buffer.
+    if (list != null) {
+      bytes.write(toIntBytes(list.size()));
+      for (ByteBuffer bb : list) {
+        addByteBuffer(bb, bytes);
+      }
+    } else {
+      bytes.write(toIntBytes(0));
+    }
+  }
+
+  public static void addByteBuffer(ByteBuffer bb,
+    ByteArrayOutputStream bytes) throws IOException {
+    if (bb != null && bytes != null) {
+      bytes.write(toIntBytes(bb.array().length));
+      bytes.write(bb.array());
+    }
+  }
+
+  public static ByteBuffer toByteBuffer(Protos.TaskID taskId) {
+    return toBuffer(taskId);
+  }
+
+  public static ByteBuffer toByteBuffer(Protos.FrameworkID frameworkId) {
+    return toBuffer(frameworkId);
+  }
+
+  /*
+   * Common method to convert Protobuf object to ByteBuffer 
+   */
+  public static ByteBuffer toBuffer(GeneratedMessage message) {
```

Comment:
The toBuffer method is called by two other methods 
public static ByteBuffer toByteBuffer(Protos.TaskID taskId)
public static ByteBuffer toByteBuffer(Protos.FrameworkID frameworkId).
The implementation is common for both, and GeneratedMessage is a base class for both TaskID and FrameworkId, so had to name it differently.